design/ceph/submariner-globalnet.md

```diff
@@ -0,0 +1,96 @@
+# Submariner Globalnet support in Rook
+
+
+## Use case
+As a rook-ceph user, I should be able to setup mirroing across multiple clusters with overlapping networks in order to safegurd my data in case of any disaster.
+
+
+## Background
+[Submariner](https://submariner.io/getting-started/) provides cross-cluster connectivity between two kubernetes clusters when they have different CIDRs. Many a times customers don't have control over the network configuration of the clusters, resulting in clusters with overlapping CIDRs (ServiceCIDR and ClusterCIDR). For such scenarios, Submariner provides another tool called [Globalnet](https://submariner.io/getting-started/architecture/globalnet/). It provides cross-cluster connectivity from pods to remote services using their global IPs.
+
+In order to support kubernetes clusters connected using Submariner-Globalnet, Rook needs to use the global IPs provided by Globalnet instead of the service cluster IP.
+
+
+## Prerequisite
+- The peer clusters should be connected using Submariner.
+- Globalnet controller should be running on both the clusters.
+(Refer Submariner [documentation](https://submariner.io/getting-started/quickstart/) for installation steps on different environments)
+
+## Service of OSD pods
+- Implement k8s service for each osd pod:
+```yaml
+apiVersion: v1
+  kind: Service
+  metadata:
+    labels:
+      app: rook-ceph-osd
+      app.kubernetes.io/component: cephclusters.ceph.rook.io
+      app.kubernetes.io/created-by: rook-ceph-operator
+      app.kubernetes.io/instance: <osd-id>
+      app.kubernetes.io/managed-by: rook-ceph-operator
+      app.kubernetes.io/name: ceph-osd
+      app.kubernetes.io/part-of: rook-ceph
+      ceph_daemon_id: <osd-id>
+      ceph_daemon_type: osd
+      osd: <osd-id>
+      osd_cluster: rook-ceph
+      rook.io/operator-namespace: rook-ceph
+      rook_cluster: rook-ceph
+    name: rook-ceph-mon-<id>
+    namespace: rook-ceph
+  spec:
+    selector:
+      app: rook-ceph-osd
+      ceph_daemon_id: <osd-id>
+      osd: <osd-id>
+      osd_cluster: rook-ceph
+      rook_cluster: rook-ceph
+    type: ClusterIP
+```
```

Comment:
I think defining something as simple as the Service here is probably a bit unnecessary. I would probably omit this.